src/Appwrite/Platform/Workers/Messaging.php

```diff
@@ -104,4 +93,71 @@ public function action(Message $message): void
             throw new Exception('Error sending message: ' . $error->getMessage(), 500);
         }
     }
+
+    protected static function createAdapterFromDSN(DSN $dsn): SMSAdapter
+    {
+        $from = empty($dsn->getParam('from', '')) ? null : $dsn->getParam('from', '');
```

Comment:
Seems odd that you can't use a null default here 🤔 let's update the DSN lib to allow a nullable default, not a blocker for this PR though